sharding/nodesCoordinator/indexHashedNodesCoordinator.go

```diff
@@ -612,33 +613,18 @@ func (ihnc *indexHashedNodesCoordinator) EpochStartPrepare(metaHdr data.HeaderHa
 		return
 	}
 
-	ihnc.mutNodesConfig.RLock()
-	previousConfig := ihnc.nodesConfig[ihnc.currentEpoch]
-	if previousConfig == nil {
-		log.Error("previous nodes config is nil")
-		ihnc.mutNodesConfig.RUnlock()
-		return
-	}
-
-	// TODO: remove the copy if no changes are done to the maps
-	copiedPrevious := &epochNodesConfig{}
-	copiedPrevious.eligibleMap = copyValidatorMap(previousConfig.eligibleMap)
-	copiedPrevious.waitingMap = copyValidatorMap(previousConfig.waitingMap)
-	copiedPrevious.nbShards = previousConfig.nbShards
-
-	ihnc.mutNodesConfig.RUnlock()
-
 	// TODO: compare with previous nodesConfig if exists
-	newNodesConfig, err := ihnc.computeNodesConfigFromList(copiedPrevious, allValidatorInfo)
+	newNodesConfig, err := ihnc.computeNodesConfigFromList(allValidatorInfo)
```

Comment:
how can this be backward compatible ? it you have no information ?

Reply:
It is backwards compatible, since the `previousConfig` was never used on mainnet. It was supposed to be enabled with `FixWaitingListEnableEpoch`, which I've deleted in a previous PR.

Old version compatibility is maintained in `addValidatorToPreviousMap` :
```go
	if !ihnc.flagStakingV4Started.IsSet() {
		eligibleMap[shardId] = append(eligibleMap[shardId], currentValidator)
		return
	}
``` 

Here, `eligibleMap` **is not** the one from `previousEpoch`, is actually the received list from peerMiniBlocks